File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0;
        int maxSum = INT_MIN;
        for(int num: nums) {
            sum = max(sum + num, num);
            maxSum = max(sum, maxSum);
        }
        return maxSum;
    }
};

File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSum = nums[0], currSum = 0;
    for(let i=0; i<nums.length; i++) {
        currSum += nums[i];
        maxSum = Math.max(maxSum, currSum);
        if(currSum < 0) {
            currSum = 0;
        }
    }
    return maxSum;
};